Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -528,22 +529,22 @@ void LocalFrameViewLayoutContext::applyTextSizingIfNeeded(RenderElement& layoutR
 void LocalFrameViewLayoutContext::updateStyleForLayout()
 {
     ScriptDisallowedScope::InMainThread scriptDisallowedScope;
-    Document& document = *frame().document();
+    Ref document = *frame().document();
 
     // FIXME: This shouldn't be necessary, but see rdar://problem/36670246.
-    if (!document.styleScope().resolverIfExists())
-        document.styleScope().didChangeStyleSheetEnvironment();
+    if (!document->styleScope().resolverIfExists())
```

Comment:
Do we need a `protectedStyleScope()`?

Reply:
1. resolverIfExists() seems like a trivial function.
2. StyleScope is not RefCounted.

We could call `checkedStyleScope()` but given that `resolverIfExists()` seems trivial, not sure it is worth the refcounting churn.